extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -276,18 +276,29 @@ private Uni<TokenVerificationResult> verifyCodeFlowAccessTokenUni(RoutingContext
                 && (resolvedContext.oidcConfig.authentication.verifyAccessToken
                         || resolvedContext.oidcConfig.roles.source.orElse(null) == Source.accesstoken)) {
             final String codeAccessToken = (String) vertxContext.get(OidcConstants.ACCESS_TOKEN_VALUE);
-            return verifyTokenUni(resolvedContext, codeAccessToken);
+            return verifyTokenUni(resolvedContext, codeAccessToken, null);
```

Comment:
Interesting, so it shows the the binary code flow access tokens can not be verified, if the provider has no introspection endpoint. There is a property, verify code flow access tokens, which can be set it true, I set to to true once by default and immediately I got users reporting problems with some of non-Keycloak providers. The reason one might want to verify such a binary code flow token is because in the code flow we only verify ID token, the access token is meant to be propagated only - so the endpoint might want to enforce its verification to avoid being a mere conduit of some binary blobs.
I think, by pushing the opaque token verification down we've found that `UserInfo` should be done earlier. I.e right now `OidcIidentityProvider` works as follows - 1) verify code flow access token if any 2) get user info if needed 3) verify the primary token - ID token in case of the code flow - or bearer access token
But it should be:
1) get user info if needed - it does the indirect verification 2) verify code flow access token if required and if it is code flow 3) verify the primary token - ID token in case of the code flow - or bearer access token
 
This will let you pass `UserInfo` instead of `null`

Reply:
When you get a chance, no rush, thanks